### ISSUE TITLE
[master] [regression] execute directive does not work with advanced bash commands

### ISSUE THREAD

**Issue by Medath** [lutris/lutris#1669]
I want to execute `for i in $GAMEDIR/drive_c/rct3/Music/*.wma; do ffmpeg -i "$i" "${i%.*}.mp3"; done` in an installer. To achieve that I came up with the following piece of code:
```
        - execute:
            description: This is a test
            command: 'for i in $GAMEDIR/drive_c/rct3/Music/*.wma; do ffmpeg -i "$i" "${i%.*}.mp3"; done'
```
In 0.4.23 this works just fine. But it fails on master.

`Running /usr/bin/bash -c for i in /mnt/7f2b9e1c-a8ef-4830-93fa-56f508d8ee98/LutrisGames/rollercoaster-tycoon-3-test/drive_c/rct3/Music/*.wma; do ffmpeg -i "$i" "${i%.*}.mp3"; done` is printed on the terminal (using master). This suggests that Lutris does not put quotation marks around the command, which are definitely needed in this case.

**Comment by Medath** [comment]
Ok, after making a quick fix that puts the command in quotes it still does not work.
```
DEBUG    2019-01-26 20:09:30,873 [interpreter._iter_commands:527]:Installer command: {'execute': {'command': 'for i in $GAMEDIR/drive_c/rct3/Music/*.wma; do ffmpeg -i "$i" "${i%.*}.mp3"; done'}}
DEBUG    2019-01-26 20:09:30,875 [command.start:112]:Running /home/username/Desktop/git-projects/lutris/real/bin/lutris-wrapper 1 0 bash /usr/bin/bash -c 'for i in /mnt/drive/LutrisGames/rollercoaster-tycoon-3-wat/drive_c/rct3/Music/*.wma; do ffmpeg -i "$i" "${i%.*}.mp3"; done'
DEBUG    2019-01-26 20:09:30,875 [command.start:114]:ENV: STEAM_RUNTIME="/home/username/.local/share/lutris/runtime/steam"
DEBUG    2019-01-26 20:09:30,875 [command.start:114]:ENV: LD_LIBRARY_PATH="/home/username/.local/share/lutris/runtime/Ubuntu-18.04-i686:/home/username/.local/share/lutris/runtime/steam/i386/lib/i386-linux-gnu:/home/username/.local/share/lutris/runtime/steam/i386/lib:/home/username/.local/share/lutris/runtime/steam/i386/usr/lib/i386-linux-gnu:/home/username/.local/share/lutris/runtime/steam/i386/usr/lib:/home/username/.local/share/lutris/runtime/Ubuntu-18.04-x86_64:/home/username/.local/share/lutris/runtime/steam/amd64/lib/x86_64-linux-gnu:/home/username/.local/share/lutris/runtime/steam/amd64/lib:/home/username/.local/share/lutris/runtime/steam/amd64/usr/lib/x86_64-linux-gnu:/home/username/.local/share/lutris/runtime/steam/amd64/usr/lib"
DEBUG    2019-01-26 20:09:30,875 [command.start:114]:ENV: PYTHONPATH="/home/username/Desktop/git-projects/lutris/real:/home/username/Desktop/git-projects/lutris/real/bin:/usr/lib/python37.zip:/usr/lib/python3.7:/usr/lib/python3.7/lib-dynload:/usr/lib/python3.7/site-packages"
DEBUG    2019-01-26 20:09:30,881 [signals.register_handler:26]:Registering <bound method MonitoredCommand.on_stop of <lutris.command.MonitoredCommand object at 0x7fc34324b1d0>> to 6562
/usr/bin/bash: for i in /mnt/drive/LutrisGames/rollercoaster-tycoon-3-wat/drive_c/rct3/Music/*.wma; do ffmpeg -i "$i" "${i%.*}.mp3"; done: No such file or directory
Running /usr/bin/bash -c 'for i in /mnt/drive/LutrisGames/rollercoaster-tycoon-3-wat/drive_c/rct3/Music/*.wma; do ffmpeg -i "$i" "${i%.*}.mp3"; done'
DEBUG    2019-01-26 20:09:31,055 [command.on_stop:151]:The process has terminated with code 32512
DEBUG    2019-01-26 20:09:31,055 [command.stop:247]:Management process looks dead already.
DEBUG    2019-01-26 20:09:31,055 [command.stop:258]:logger already detached
DEBUG    2019-01-26 20:09:31,055 [command.restore_environment:225]:Restoring environment
```

Just to make it clear, manually executing `/usr/bin/bash -c 'for i in /mnt/drive/LutrisGames/rollercoaster-tycoon-3-wat/drive_c/rct3/Music/*.wma; do ffmpeg -i "$i" "${i%.*}.mp3"; done'` in a terminal works without issues. I also don't encounter any of these problems on 0.4.23. So the problem has to be in Lutris somewhere.

**Comment by strycore** [comment]
Can you test this and confirm that is doesn't introduce other issues?

**Comment by Medath** [comment]
This had no effect for me.
It seems like the issue is not really related to the process monitor.

With some testing I found that running a *simple* command *without* quotes works fine (I used `sleep 600` in my tests), even before 6cf2edb. If my command wouldn't require to be in quotes, it would probably run fine as well.

When I put quotes around any command (either by using quotes in my yaml file or by patching lutris to put quotes around it), things start to get weird. As can be seen in the output of my second comment, there is this interesting line: `/usr/bin/bash: for i in /mnt/drive/LutrisGames/rollercoaster-tycoon-3-wat/drive_c/rct3/Music/*.wma; do ffmpeg -i "$i" "${i%.*}.mp3"; done: No such file or directory`
This can be replicated outside of Lutris by omitting the -c parameter when running the command with bash. But, according to the terminal output of Lutris, the -c parameter is definitely used. What's weird as well is that this error message is printed *before* Lutris says it's about to run that command (as seen in my second comment).

This entire stuff can be worked around by using the `file` and `args` parameters instead of the `command` parameter. So my working solution is now this:
```
- execute:
    args: '-c ''for i in $GAMEDIR/drive_c/rct3/Music/*.wma; do ffmpeg -i "$i" "${i%.*}.mp3"; done'''
    file: bash
```

TL;DR: No, this did not fix my issue unfortunately. But I found a workaround for now. I'll investigate more

**Comment by strycore** [comment]
This is not a workaround
This is how the command is intended to be used.
Leaving the issue closed and considered solved
